core/server/api/authentication.js

```diff
@@ -260,22 +272,21 @@ authentication = {
         });
     },
 
-    revoke: function (object) {
-        var token;
-
-        if (object.token_type_hint && object.token_type_hint === 'access_token') {
-            token = dataProvider.Accesstoken;
-        } else if (object.token_type_hint && object.token_type_hint === 'refresh_token') {
-            token = dataProvider.Refreshtoken;
-        } else {
-            return errors.BadRequestError('Invalid token_type_hint given.');
+    updateSetup: function updateSetup(object, options) {
+        if (!options.context || !options.context.user) {
```

Comment:
As far as I am aware, this line is always going to get triggered in use over HTTP. The reason being that `context.user` comes from `req.user`, which is being skipped for the auth route: https://github.com/TryGhost/Ghost/blob/master/core/server/middleware/middleware.js#L114

We need to make sure that the `PUT /authentication/setup/`route is not skipped by auth. We have access to req.method to add this extra logic to the middleware.